bin/brew

```diff
@@ -41,149 +18,9 @@ HOMEBREW_PREFIX="$(chdir "$(dirname "$(dirname "$HOMEBREW_BREW_FILE")")" && pwd
 HOMEBREW_REPOSITORY="$(chdir "$BREW_FILE_DIRECTORY"/../ && pwd -P)"
 HOMEBREW_LIBRARY="$(chdir "$BREW_FILE_DIRECTORY"/../Library && pwd -P)"
 
-# Where we store built products; /usr/local/Cellar if it exists,
-# otherwise a Cellar relative to the Repository.
-if [[ -d "$HOMEBREW_PREFIX/Cellar" ]]
-then
-  HOMEBREW_CELLAR="$HOMEBREW_PREFIX/Cellar"
-else
-  HOMEBREW_CELLAR="$HOMEBREW_REPOSITORY/Cellar"
-fi
-
-case "$*" in
-  --prefix) echo "$HOMEBREW_PREFIX"; exit 0 ;;
-  --cellar) echo "$HOMEBREW_CELLAR"; exit 0 ;;
-  --repository|--repo) echo "$HOMEBREW_REPOSITORY"; exit 0 ;;
-esac
-
-if [[ "$HOMEBREW_PREFIX" = "/" || "$HOMEBREW_PREFIX" = "/usr" ]]
-then
-  # it may work, but I only see pain this route and don't want to support it
-  odie "Cowardly refusing to continue at this prefix: $HOMEBREW_PREFIX"
-fi
-
-# Users may have these set, pointing the system Ruby
-# at non-system gem paths
-unset GEM_HOME
-unset GEM_PATH
-
-if [[ -z "$HOMEBREW_DEVELOPER" ]]
-then
-  unset HOMEBREW_RUBY_PATH
-fi
-
-if [[ "$(uname -s)" = "Darwin" ]]
-then
-  HOMEBREW_OSX="1"
-fi
-
-if [[ -z "$HOMEBREW_RUBY_PATH" ]]
-then
-  if [[ -n "$HOMEBREW_OSX" ]]
-  then
-    HOMEBREW_RUBY_PATH="/System/Library/Frameworks/Ruby.framework/Versions/Current/usr/bin/ruby"
-  else
-    HOMEBREW_RUBY_PATH="$(which ruby)"
-  fi
-fi
-
 export HOMEBREW_BREW_FILE
-export HOMEBREW_RUBY_PATH
 export HOMEBREW_PREFIX
 export HOMEBREW_REPOSITORY
 export HOMEBREW_LIBRARY
-export HOMEBREW_CELLAR
```

Comment:
If we source `Library/brew.sh` instead of executing it in a new shell, these `export`s can be moved there, too.